rewrite-graphql/src/main/java/org/openrewrite/graphql/internal/GraphQlPrinter.java

```diff
@@ -0,0 +1,1083 @@
+            }
+        }
+        
+        p.append("}");
+        afterSyntax(schemaDef, p);
+        return schemaDef;
+    }
+
+    @Override
+    public GraphQl visitRootOperationTypeDefinition(GraphQl.RootOperationTypeDefinition rootOpType, PrintOutputCapture<P> p) {
+        beforeSyntax(rootOpType, p);
+        p.append(rootOpType.getOperationType().name().toLowerCase());
+        p.append(":");
+        visitNamedType(rootOpType.getType(), p);
+        afterSyntax(rootOpType, p);
+        return rootOpType;
+    }
+
+    @Override
+    public GraphQl visitScalarTypeDefinition(GraphQl.ScalarTypeDefinition scalarType, PrintOutputCapture<P> p) {
+        beforeSyntax(scalarType, p);
+        if (scalarType.getDescription() != null) {
+            visitStringValue(scalarType.getDescription(), p);
+            p.append(" ");
+        }
+        p.append("scalar");
+        visitName(scalarType.getName(), p);
+        visitList(scalarType.getDirectives(), p);
+        afterSyntax(scalarType, p);
+        return scalarType;
+    }
+
+    @Override
+    public GraphQl visitObjectTypeDefinition(GraphQl.ObjectTypeDefinition objectType, PrintOutputCapture<P> p) {
+        beforeSyntax(objectType, p);
+        if (objectType.getDescription() != null) {
+            visitStringValue(objectType.getDescription(), p);
+            p.append(" ");
+        }
+        p.append("type");
+        visitName(objectType.getName(), p);
+        visitImplementsInterfaces(objectType.getImplementsInterfaces(), p);
+        visitList(objectType.getDirectives(), p);
+        if (objectType.getFieldsPrefix() != null) {
+            printSpace(objectType.getFieldsPrefix(), p);
+            if (objectType.getFields() != null && !objectType.getFields().isEmpty()) {
+                visitContainer("{", objectType.getFields(), "}", null, p);
+            } else {
+                p.append("{}");
+            }
+        }
+        afterSyntax(objectType, p);
+        return objectType;
+    }
+
+    @Override
+    public GraphQl visitFieldDefinition(GraphQl.FieldDefinition fieldDef, PrintOutputCapture<P> p) {
+        beforeSyntax(fieldDef, p);
+        if (fieldDef.getDescription() != null) {
+            visitStringValue(fieldDef.getDescription(), p);
+            p.append(" ");
+        }
+        visitName(fieldDef.getName(), p);
+        visitInputValueDefinitions(fieldDef.getArguments(), fieldDef.getArgumentsEnd(), p);
+        p.append(":");
+        visitType(fieldDef.getType(), p);
+        visitList(fieldDef.getDirectives(), p);
+        afterSyntax(fieldDef, p);
+        return fieldDef;
+    }
+
+    @Override
+    public GraphQl visitInputValueDefinition(GraphQl.InputValueDefinition inputValueDef, PrintOutputCapture<P> p) {
+        beforeSyntax(inputValueDef, p);
+        if (inputValueDef.getDescription() != null) {
+            visitStringValue(inputValueDef.getDescription(), p);
+            p.append(" ");
```

Comment:
Printing single space literals like this is always fishy. Probably means what we should look into making something a `LeftPadded` or `RightPadded` somewhere.